podcasts/Onboarding/Plus/UpgradeLandingView.swift

```diff
@@ -0,0 +1,324 @@
+import SwiftUI
+
+struct UpgradeLandingView: View {
+    @EnvironmentObject var viewModel: PlusLandingViewModel
+
+    private let tiers: [UpgradeTier] = [.plus, .patron]
+
+    @State private var currentPage: Int = 0
+
+    @State private var displayPrice: DisplayPrice = .yearly
+
+    var body: some View {
+        VStack(spacing: 0) {
+            topBar
+
+            ZStack {
+
+                if currentPage == 0 {
+                    tiers[0].background
+                        .transition(.opacity.animation(currentPage == 0 ? .easeIn : .easeOut))
+                        .ignoresSafeArea()
+                } else {
+                    tiers[1].background
+                        .transition(.opacity.animation(currentPage == 0 ? .easeIn : .easeOut))
+                        .ignoresSafeArea()
+                }
+
+                VStack(spacing: 0) {
+                    ScrollViewIfNeeded {
+                        VStack(spacing: 0) {
+
+
+                            PlusLabel(currentPage == 0 ? L10n.plusMarketingTitle : L10n.patronCallout, for: .title2)
+                                .minimumScaleFactor(0.5)
+                                .lineLimit(2)
+                                .padding(.bottom, 16)
+                                .padding(.horizontal, 32)
+
+                            UpgradeRoundedSegmentedControl(selected: $displayPrice)
+                                .padding(.bottom, 24)
+
+                            FeaturesCarousel(currentIndex: $currentPage.animation(), currentPrice: $displayPrice, tiers: tiers)
+
+                            PageIndicator(numberOfItems: tiers.count, currentPage: currentPage)
+                            .padding(.top, 27)
+                        }
+                    }
+                }
+            }
+        }
+        .background(Color(hex: "121212"))
+    }
+
+    var topBar: some View {
+        HStack(spacing: 0) {
+            Rectangle()
+                .foregroundColor(.clear)
+                .frame(height: 44)
+            Button(viewModel.source == .upsell ? L10n.eoyNotNow : L10n.plusSkip) {
+                viewModel.dismissTapped()
+            }
+            .foregroundColor(.white)
+            .font(style: .body, weight: .medium)
+            .padding(.trailing)
+        }
```

Comment:
wdyt about restructuring this so button has more tappable space by shifting the padding to the button instead of using the rectangle to define the height?

before (top) - after (bottom): 
<img width="320" alt="Screen Shot 2023-04-12 at 12 05 29 PM" src="https://user-images.githubusercontent.com/793774/231516820-2839a270-8d8f-44d2-9889-1b1094e4d674.png">

```suggestion
            HStack(spacing: 0) {
                Spacer()

                Button(viewModel.source == .upsell ? L10n.eoyNotNow : L10n.plusSkip) {
                    viewModel.dismissTapped()
                }
                .foregroundColor(.white)
                .font(style: .body, weight: .medium)
                .padding()
            }
```

Reply:
I tried that, but it seems weird on smaller iPhones. Anyway, the changes on this PR are now making the "Dismiss/Not Now" to be out of the screen on smaller iPhones. I'll move with them anyway and look at this later. Changed in 6e62ecaeec36228b3d656f9742343f74ef87f5d7